pandas/core/dtypes/concat.py

```diff
@@ -111,11 +111,15 @@ def is_nonempty(x) -> bool:
         to_concat = non_empties
 
     kinds = {obj.dtype.kind for obj in to_concat}
+    _contains_datetime = any(kind in ["m", "M"] for kind in kinds)
```

Comment:
not sure much point in making this private

Reply:
sure